mlir-sycl/include/mlir/Dialect/SYCL/IR/SYCLOps.td

```diff
@@ -130,8 +130,9 @@ class SYCLInheritanceTypeTrait<string ParentClass>
 
 def SYCL_IDType
     : SYCL_Type<"ID", "id", [SYCLInheritanceTypeTrait<"ArrayType">]> {
-  let parameters = (ins "unsigned":$dimension);
-  let assemblyFormat = "`<` $dimension `>`";
+  let parameters = (ins "unsigned":$dimension,
+                        "llvm::SmallVector<mlir::Type, 4>":$body);
```

Comment:
As I'll merge the commit now, you should use ArrayRefParameter<"mlir::Type"> here